tests/test_base_client.py

```diff
@@ -276,3 +277,38 @@ def inner(h: BaseHTTPRequestHandler):
         assert 'foo' in res
 
     assert len(requests) == 2
+
+
+@pytest.mark.parametrize('chunk_size,expected_chunks,data_size',
+                         [(5, 20, 100), # 100 / 5 bytes per chunk = 20 chunks
+                          (10, 10, 100), # 100 / 10 bytes per chunk = 10 chunks
+                          (200, 1, 100), # 100 / 200 bytes per chunk = 1 chunk
+                          ])
+def test_streaming_response_chunk_size(chunk_size, expected_chunks, data_size):
+    test_data = b"0" * data_size
```

Comment:
Does it make sense to generate a variable test data (e.g. random with the fixed seed or a hard-coded block of text) so we have a higher confidence that bytes/chunks did not get mixed?

Reply:
Agreed, done.